web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx

```diff
@@ -97,14 +98,9 @@ export function EnrollEksCluster(props: AgentStepProps) {
     useState(false);
   const [isManualHelmDialogShown, setIsManualHelmDialogShown] = useState(false);
   const [waitingResourceId, setWaitingResourceId] = useState('');
+  const [joinToken, setJoinToken] = useState<JoinToken>(null);
```

Comment:
Let's document the fact that we need a separate state for the join token because while the join token is used mostly by code in `EnrollEksCluster`, the join token needs to be generated only after `ManualHelmDialog` gets open.

If I saw this code without any context, my inclination would be to move `useJoinTokenSuspender` to `EnrollEksCluster` and pass the token down to `ManualHelmDialog`, because that's common practice in React – if you want to share a piece of state, you move it up. Here it wouldn't work of course because of the constraints you talk about in the PR description.